lib/decode_session_cookie.rb

```diff
@@ -0,0 +1,53 @@
+# decode_session_cookie.rb
+# ------------------------
+# The purpose of this script is to show that if I have the secret_key_base
+# and a cookie to an active Rails session, I can decrypt it and gain access
+# to key information about the user's session.
+
+require 'rubygems'
+require 'cgi'
+require 'active_support'
+
+def decrypt_session_cookie(cookie, key)
+  cookie = CGI::unescape(cookie)
+
+  # Default values for Rails 4 apps
+  key_iter_num = 1000
+  key_size     = 64
+  salt         = "encrypted cookie"
+  signed_salt  = "signed encrypted cookie"
+
+  key_generator = ActiveSupport::KeyGenerator.new(key, iterations: key_iter_num)
+  secret = key_generator.generate_key(salt)
+  sign_secret = key_generator.generate_key(signed_salt)
+
+  encryptor = ActiveSupport::MessageEncryptor.new(secret, sign_secret)
+  encryptor.decrypt_and_verify(cookie)
+end
+
+def unmask_authenticity_token(token, length = 32)
+  masked_token = Base64.strict_decode64(token)
+  # Split the token into the one-time pad and the encrypted
+  # value and decrypt it
+  one_time_pad = masked_token[0...length]
+  encrypted_csrf_token = masked_token[length..-1]
+  xor_byte_strings(one_time_pad, encrypted_csrf_token)
+end
+
+def xor_byte_strings(s1, s2)
+  s1.bytes.zip(s2.bytes).map { |(c1,c2)| c1 ^ c2 }.pack('c*')
```

Comment:
Space missing after comma.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.